src/Debugging-Core-Tests/DebuggingInstructionStreamTest.class.st

```diff
@@ -0,0 +1,141 @@
+Class {
+	#name : #DebuggingInstructionStreamTest,
+	#superclass : #TestCase,
+	#category : #'Debugging-Core-Tests'
+}
+
+{ #category : #helpers }
+DebuggingInstructionStreamTest >> method1 [
+
+	| a |
+	a := 1.
+	^ 1 + 2
+]
+
+{ #category : #helpers }
+DebuggingInstructionStreamTest >> methodWithBlock [
+
+	[| a |
+	a := 1.
+	^ 1 + 2] value
+]
+
+{ #category : #helpers }
+DebuggingInstructionStreamTest >> methodWithSuper [
+
+	super printString.
+	^ 42
+]
+
+{ #category : #'tests - structure' }
+DebuggingInstructionStreamTest >> testFirstByte [
+	|compiledMethod stream|
+	compiledMethod := self class >> #method1.
+	stream := InstructionStream on: compiledMethod.
+	self assert: stream firstByte equals: (compiledMethod at: compiledMethod initialPC)
+]
+
+{ #category : #'tests - structure' }
+DebuggingInstructionStreamTest >> testFollowingByteAccessByteCodeConstituentByteByByte [
+
+	| compiledMethod stream client |
+	compiledMethod := self class >> #methodWithSuper.
+	stream := InstructionStream on: compiledMethod.
+	client := InstructionClient new.
+	
+	"Answer the next byte.
+	For example for this bytecode 
+	<EB 00> superSend 
+	<D8> pop
+	 If the current byte is EB, the following byte is 00.
+	 It is not the following bytecode, that is D8"
+ 	stream pc: stream pc + 1.	
```

Comment:
I don't understand why the +1 here. It is assuming 00? But by reading the comment I would have thought your stream is at EB.

Reply:
Here we wanted to show is that if we have a bytecode with two bytes, the followingByte (not bytecode) is the one following the one pointed by the current PC
So we point at EB and followingByte should be 00. 
May be the test is wrong this is what we want to test.